pkg/controller/machinedeployment/machinedeployment_controller_test.go

```diff
@@ -153,7 +154,16 @@ func TestReconcile(t *testing.T) {
 	// Wait for the new MachineSet to get scaled up and set .Status.Replicas and .Status.AvailableReplicas
 	// at each step
 	var newMachineSet, oldMachineSet *clusterv1alpha1.MachineSet
-	if machineSets.Items[0].CreationTimestamp.Before(&machineSets.Items[1].CreationTimestamp) {
+	resourceVersion0, err0 := strconv.Atoi(machineSets.Items[0].ResourceVersion)
+	resourceVersion1, err1 := strconv.Atoi(machineSets.Items[1].ResourceVersion)
```

Comment:
While this works today is it guaranteed to work in the future? I thought comparing resourceVersions was only expected to work against the same instance of a resource and not across different instances of a resource

Reply:
Is not the new MS created from the old MS? Something that we could assume at least on the unit tests?

Reply:
After digging into this a bit more, this behavior appears to be safe and is also the way that Deployments interact with ReplicaSets.